shub/config.py

```diff
@@ -164,15 +171,47 @@ def get_version(self):
         elif self.version:
             return str(self.version)
 
+    def get_stack(self, target):
+        return self.stacks.get(target)
+
     def get_target(self, target, auth_required=True):
         """Return (project_id, endpoint, apikey) for given target."""
-        return (
-            self.get_project_id(target),
-            self.get_endpoint(target),
-            self.get_apikey(target, required=auth_required),
+        return Target(
+            name=target,
+            project_id=self.get_project_id(target),
+            endpoint=self.get_endpoint(target),
+            apikey=self.get_apikey(target, required=auth_required),
+            stack=self.get_stack(target),
+            requirements_file=self.requirements_file,
         )
 
 
+class Target(object):
+
+    def __init__(self, name, project_id, endpoint, apikey, stack,
+                 requirements_file):
+        self.name = name
+        self.project_id = project_id
+        self.endpoint = endpoint
+        self.apikey = apikey
+        self.stack = stack
+        self.requirements_file = requirements_file
+
+    def __getitem__(self, pos):
+        return (self.project_id, self.endpoint, self.apikey)[pos]
```

Comment:
This looks like a backwards-compatibility layer for my older screwed-up implementation (should have used `namedtuple` instead of `tuple` in the first place). I would rather scratch this and seize the chance to clean up the codebase, using `target.project_id` instead of `target[0]` everywhere.